media.go

```diff
@@ -551,10 +593,20 @@ func (ms *mediaSource[T, U]) Stream(ctx context.Context, errHandlers ...ErrorHan
 	defer prodCon.stateMu.Unlock()
 
 	if currentSpan := trace.FromContext(ctx); currentSpan != nil {
-		prodCon.cancelCtx = trace.NewContext(prodCon.cancelCtx, currentSpan)
+		func() {
+			prodCon.cancelCtxMu.Lock()
+			defer prodCon.cancelCtxMu.Unlock()
+			prodCon.cancelCtx = trace.NewContext(prodCon.cancelCtx, currentSpan)
+		}()
 	}
 
-	cancelCtx, cancel := context.WithCancel(prodCon.cancelCtx)
+	var cancelCtx context.Context
+	var cancel context.CancelFunc
+	func() {
+		prodCon.cancelCtxMu.RLock()
+		cancelCtx, cancel = context.WithCancel(prodCon.cancelCtx)
+		prodCon.cancelCtxMu.RUnlock()
```

Comment:
This is a total nit, but is there a reason this isn't deferred like the other locks?

Reply:
Good catch! Done.